example/service/rds/rdsutils/authentication/iam_authentication.go

```diff
@@ -36,7 +37,7 @@ func main() {
 
 	credProvider := stscreds.NewAssumeRoleProvider(sts.New(cfg), os.Args[5])
 
-	authToken, err := rdsutils.BuildAuthToken(dbEndpoint, awsRegion, dbUser, credProvider)
+	authToken, err := rdsutils.BuildAuthToken(context.Background(), dbEndpoint, awsRegion, dbUser, credProvider)
```

Comment:
If we're updateing this `BuildAuthToken` and similar utilities to take a signer instead of credential provider, we'll still want to updates these examples on how to create a signer for the credProvider.

Reply:
will update BuildAuthToken to take a pointer to the signer.